Update i18n docs to reflect new nav label translation system (#35)
withastro/contribute.docs.astro.build: src/content/docs/guides/i18n.mdx

# ...
```

See the [English](https://github.com/withastro/docs/tree/main/src/content/i18n/en.yml) or [French](https://github.com/withastro/docs/tree/main/src/content/i18n/fr.yml) files in the docs repo as examples of a what a full file looks like.

#### Sidebar navigation labels

> Similar to reusable UI strings, we have labels for the groups displayed in the docs sidebar. For example, the labels for the sidebar tabs are stored in these files.

Reviewer: ```suggestion
Similar to reusable UI strings, the main navigation headings displayed in the docs sidebar are translated in their own file.
```
The for example felt a little redundant maybe? I think it's OK just to say in an opening sentence these things ARE translated. Then below describes how.

Author: I was avoiding “headings” because this includes stuff like the tabs which I’m not sure I’d describe as “headings”? I guess we can adjust your suggestion like this:
```suggestion
Similar to reusable UI strings, labels for the groups displayed in the docs sidebar are translated in their own file.
```

> Navigation labels are stored in the `src/content/nav/` folder, with a TypeScript file for each language. The non-English files should contain a default export using the `navDictionary()` helper method. Just like for UI strings, keys are not translated, only the values in the object passed to `navDictionary()`.

Reviewer: ```suggestion
Navigation labels are stored in the `src/content/nav/` folder, with a TypeScript file for each language. The `en.ts` file in this folder contains the English text displayed in the sidebar for each key.

The non-English files must contain a default export using the `navDictionary()` helper method. Just like for UI strings, keys are not translated, only the values in the object passed to `navDictionary()`.

Note that the values for your language are not simply translations of the key words themselves. For example, The `start.welcome` key has a value of `'Welcome, world!"` to translate. You will have to consult the `en.ts` file in order to translate these values for each key. 
```
Is this helpful, or unnecessary? Just a thought I had to prompt people early to check the `en` file, and I had visions of people starting with "OK, start is X. Moving on to "welcome"... but maybe that's unwarranted!

Author: I like mentioning `en.ts`, but I think the third paragraph is probably overkill.

```suggestion
Navigation labels are stored in the `src/content/nav/` folder, with a TypeScript file for each language. The `en.ts` file in this folder contains the English text displayed in the sidebar for each key.

The non-English files must contain a default export using the `navDictionary()` helper method. Just like for UI strings, keys are not translated, only the values in the object passed to `navDictionary()`.
```
